packages/data-table-server/src/app/createApp.ts

```diff
@@ -2,14 +2,24 @@
  * Tupaia
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
+import { ForwardingAuthHandler } from '@tupaia/api-client';
 import { TupaiaDatabase } from '@tupaia/database';
-import { MicroServiceApiBuilder } from '@tupaia/server-boilerplate';
+import { handleWith, MicroServiceApiBuilder } from '@tupaia/server-boilerplate';
+
+import { FetchDataTableDataRequest, FetchDataTableDataRoute } from '../routes';
 
 /**
  * Set up express server with middleware,
  */
 export function createApp(database = new TupaiaDatabase()) {
-  const app = new MicroServiceApiBuilder(database, 'data-table').useBasicBearerAuth().build();
+  const app = new MicroServiceApiBuilder(database, 'data-table')
+    .attachApiClientToContext(req => new ForwardingAuthHandler(req.headers.authorization))
+    .useBasicBearerAuth()
+    .post<FetchDataTableDataRequest>(
```

Comment:
Decided to go with `POST` only for this endpoint. There is a rationale, here me out:
1. We have to use `POST` sometimes, since we may be passing in a large or nested query arguments, which can't be done when passing in as URL params
2. In code instances we always use `POST` for the above reason as it's safer, so the `GET` is just for developer convenience
3. Query params always come as strings and have to be parsed, whereas `POST` body is nicely formatted JSON already. Especially since data-tables will have dynamic parameters, it's hard to know which query params to parse as what type (string, string[], boolean, etc.)

Reply:
Wouldn't a FetchDataTableDataRequest simply refer to the data table by code? Seems like that's the implementation of the route. So GET should be sufficient? Or are you imagining we will have some data tables in the future which will need to be heavily call-time configurable?

Reply:
Discussed, may have a bunch more query params in the future, all good.